htdocs/compta/facture/class/facture.class.php

```diff
@@ -532,7 +532,7 @@ public function create(User $user, $notrigger = 0, $forceduedate = 0)
 			return -3;
 		}
 		$soc = new Societe($this->db);
-		$result = $soc->fetch($this->socid);
+		$result = $soc->fetch($this->fk_soc);
```

Comment:
The fetch_thirdparty function is based on fk_soc, so it can't fetch and the thirdparty remains empty if you test on an invoice.
With my correction, it works. 
By the way, I don't understand the choice to have deprecated fk_soc, socid seems less meaningful to me, the “fk” is used throughout the dolibarr architecture, including in the database, and it's the programming convention for foreign keys.

Reply:
This looks like a job for the DolDeprecationHandler.

If @eldy agrees, I would suppress 'public $fk_soc' in CommonInvoice' (=delete) and

```
require_once DOL_DOCUMENT_ROOT.'/core/class/doldeprecationhandler.class.php';
//...

    use DolDeprecationHandler;

    /**
     * Provide list of deprecated properties and replacements
     *
     * @return array<string,string>
     */
    protected function deprecatedProperties()
    {
        return array(
            'fk_soc' => 'socid',
        );
    }
```
See  htdocs/core/class/commonobject.class.php for reference.

Reply:
Indeed, what you say would solve the problem.
However, I still don't understand who decided to deprecate fk_soc for the reasons given above.